File: 面试题17.16按摩师.py
class Solution:
    def massage(self, nums: List[int]) -> int:
        if nums == []:
            return 0
        f = [[0, nums[0]]]
        for i in range(1, len(nums)):
            f.append([max(f[i - 1][0], f[i - 1][1]), nums[i] + f[i - 1][0]])
        return max(f.pop())

File: 面试题17.16按摩师.c
int massage(int* nums, int numsSize){
    if(numsSize == 0)return 0;
    int tag[2] = {0, nums[0]};
    int temp = tag[0];
    for(int i = 1; i < numsSize; i++){
        tag[0] = tag[0] > tag[1] ? tag[0] : tag[1];
        tag[1] = nums[i] + temp;
        temp = tag[0];
    }   
    return tag[0] > tag[1] ? tag[0] : tag[1];
}